nltk/twitter/twitterclient.py

```diff
@@ -0,0 +1,457 @@
+        :param bool to_screen: If ``True``, display the tweet texts on the screen,\
+        otherwise print to a file
+        :param bool stream: If ``True``, use the live public stream,\
+        otherwise search past public tweets
+        :param int limit: Number of tweets to process
+        :param tuple date_limit: The date at which to stop collecting new\
+        data. This should be entered as a tuple which can serve as the\
+        argument to `datetime.datetime`. E.g. `data_limit=(2015, 4, 1, 12,\
+        40)` for 12:30 pm on April 1 2015.\
+        Note that, in the case of streaming, it is the maximum date, i.e.\
+        a date in the future; if not, it is the minimum date, i.e. a date\
+        in the past
+        :param str lang: language
+        :param int retries_after_twython_exception: number of retries when\
+        searching tweets before raising an exception
+        """
+        if to_screen:
+            handler = TweetViewer(limit=limit, date_limit=date_limit)
+        else:
+            handler = TweetWriter(limit=limit, date_limit=date_limit,
+                                  stream=stream, repeat=False)
+
+        if stream:
+            self.streamer.register(handler)
+            if keywords == '' and follow == '' and locations == None:
+                self.streamer.sample()
+            else:
+                self.streamer.filter(track=keywords, follow=follow,
+                                     lang=lang, locations=locations)
+        else:
+            self.query.register(handler)
+            if keywords == '':
+                raise ValueError("Please supply at least one keyword to search for.")
+            else:
+                self.query.search_tweets(keywords, count=limit, lang=lang,
+                                         retries_after_twython_exception= \
+                                         retries_after_twython_exception)
+
+
+
+class TweetViewer(TweetHandlerI):
+    """
+    Handle data by sending it to the terminal.
+    """
+
+    def handle(self, data):
+        """
+        Direct data to `sys.stdout`
+
+        :return: return ``False`` if processing should cease, otherwise return ``True``.
+        :rtype: bool
+        :param data: Tweet object returned by Twitter API
+        """
+        text = data['text']
+        print(text)
+        self.counter += 1
+        if self.counter >= self.limit:
+            # Tell the client to disconnect
+            return False
+        return True
+
+
+class TweetWriter(TweetHandlerI):
+    """
+    Handle data by writing it to a file.
+    """
+    def __init__(self, limit=2000, date_limit=None, repeat=True, stream=True,
+                 fprefix='tweets', subdir='twitter-files'):
+        """
+        :param int limit: number of data items to process in the current\
+        round of processing
+
+        :param bool repeat: flag to determine whether multiple files should be\
```

Comment:
@ewan-klein added option to compress for `json2csv` functions in PR #1028, using gzip.

For `TweetWriter` I did some trials but faced compatibility problems in 2.7 and 3.4, and anyhow this will probably change if converting into a generator as per previous comment?